### ISSUE TITLE
Sifr icons not visible in dark mode

### ISSUE THREAD

**Issue by foss-** [neooffice/NeoOffice#5]
macOS 12.3.1
NeoOffice 2022 Professional Edition

Sifr icons not visible in dark mode:

![Screenshot 2022-05-16 at 17 05 34](https://user-images.githubusercontent.com/5708172/168624271-b82db1d4-3ebd-4664-ba8d-0dfa35a779fa.png)

**Comment by plubius** [comment]
The fix for this bug is in the NeoOffice 2022 Test 3 test patches

**Comment by foss-** [comment]
Downloaded Test 3 from https://github.com/neooffice/NeoOffice/releases (wasn't aware test builds were available inside of the current release - would it make sense to publish those as separate pre-releases on GitHub instead?).
Could not install the test build. Enough disk space available:
![hm](https://user-images.githubusercontent.com/5708172/171623716-b0e7fd0f-1bb8-4b51-8656-daba8cc7a42c.png)

**Comment by plubius** [comment]
From your screen snapshot, you are trying to install the test patch for the Mac App Store but your first post says you are running Professional Edition so try installing the <a href="https://github.com/neooffice/NeoOffice/releases/download/NeoOffice-2022/NeoOffice-2022_Professional_Edition-Patch-0-Test-3-Universal.dmg">test patch for the NeoOffice 2022 Professional Edition</a>.

**Comment by foss-** [comment]
You are correct. Although I find the naming a bit confusing at https://github.com/neooffice/NeoOffice/releases/tag/NeoOffice-2022

How would a user distinguish MAS vs Standalong in that naming scheme?

Installed the Pro edition you linked but sifr is still showing like this:
![sifr](https://user-images.githubusercontent.com/5708172/171646554-afab476c-df5f-4a4c-906a-120ba2ea6dab.png)

Hm about says `Patch 0`.  Sounds wrong. Re-downloaded `NeoOffice-2022_Professional_Edition-Patch-0-Test-3-Universal.dmg` but the pkg already shows as `Patch 0`.

Also about is hard to read in dark mode. Do you want a separate issue for that?
![about](https://user-images.githubusercontent.com/5708172/171647162-17f7a84d-3dfa-41fa-a87b-ff8aba72ff9d.png)

**Comment by plubius** [comment]
I found LibreOffice code that is setting About dialog text to light gray. For some reason, the code overrode the application default colors by hardcoding the light gray text color so I removed that code.

Also, I've been using the Sifr icon theme for the last week and I found that with my last fix, it is still difficult to see when the Save icon changed from disabled to enabled. Since the Sifr icons are unfortunately all hardcoded, medium gray bitmaps, I experimented with setting the window background color to various system colors and I found enabled Sifr icons most readable in Dark Mode when the window background is nearly black.

My latest fix is definitely a hack. Disabled icons are still difficult to read and the fix definitely gives NeoOffice a "high contrast" look when using the Sifr in Dark Mode, but it might work until someone comes up with a better fix or figures out how to mass convert the thousands of Sifr icons to a much lighter shade of gray (maybe someone knows how do that in ImgeMagick?).

Anyway, below are links to the test patches with the About dialog fix and the hacky window background fix: 

- [NeoOffice 2022 Mac App Store](https://github.com/neooffice/NeoOffice/releases/download/NeoOffice-2022/NeoOffice-2022-Patch-0-Test-4-Universal.dmg)
- [NeoOffice 2022 Professional Edition](https://github.com/neooffice/NeoOffice/releases/download/NeoOffice-2022/NeoOffice-2022_Professional_Edition-Patch-0-Test-4-Universal.dmg)

**Comment by foss-** [comment]
Thanks for not giving up on this (I thought you did when you unassigned from the issue).

Ugh, medium grey hardcoded? Sounds like a weird design choice. That makes background color selection very narrow as anything not close to black will result in too little contrast.

In [LibreOffice main](https://dev-builds.libreoffice.org/daily/master/current.html) there are light and dark icon themes for Sifr as well as SVG variants of dark and light icon theme. Would that maybe be useful to utilize Sifr in a more flexible manner?

This is after the most recent patch.

About:
![about](https://user-images.githubusercontent.com/5708172/173423316-c6ea59d0-e17b-4c7a-b4fe-4f315a85891b.png)

Sifr Icons:
![Sifr](https://user-images.githubusercontent.com/5708172/173423391-0ff5795c-5a49-424f-8023-833b7073b916.png)

**Comment by plubius** [comment]
I spent some time experimenting with ImageMagick in an attempt to whiten the Sifr icons so that they would be visible without having to make the window background color nearly black. Unfortunately, whitening worked for many icons, but for many others (like in Writer 's Table toolbar) whitening resulted in a nearly white square.

However, darkening the Sifr icons, though not as readable as whitened icons, seems to increase readability without having to make the window background color nearly black and without losing the fine lines and shading that many icons have.

Below are links to the test patches with nearly black Sifr icons:

- [NeoOffice 2022 Mac App Store](https://github.com/neooffice/NeoOffice/releases/download/NeoOffice-2022/NeoOffice-2022-Patch-0-Test-5-Universal.dmg)
- [NeoOffice 2022 Professional Edition](https://github.com/neooffice/NeoOffice/releases/download/NeoOffice-2022/NeoOffice-2022_Professional_Edition-Patch-0-Test-5-Universal.dmg)

**Comment by foss-** [comment]
For reference this is after latest patches:
![dark](https://user-images.githubusercontent.com/5708172/174284672-f21cbe22-5a18-459c-a949-097a982fd78f.png)
![light](https://user-images.githubusercontent.com/5708172/174284678-278374b4-da4a-4ac2-9eb9-3fcd1a387da8.png)

**Comment by plubius** [comment]
The darkened Sifr icons change is included in NeoOffice 2022.1:
- [NeoOffice 2022.1 Professional Edition](https://github.com/neooffice/NeoOffice/releases/latest)
- [NeoOffice 2022.1 in the Mac App Store](https://apps.apple.com/app/neooffice/id639210716?mt=12)

**Comment by foss-** [comment]
Do you want to close this bug here then? Implemented a proper dark mode solution will problem require bumping base LibreOffice version.

In any case, both light and dark mode sifr have improved with your changes.
